src/App.tsx

```diff
@@ -5,7 +5,11 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const addNewMovie = (movie: Movie) => {
+    setMovies([...movies, movie]);
```

Comment:
use callback when you use the previous state
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state